### PR TITLE
added a  method to get the child organizations

### PR DESCRIPTION
**Pull request by satti-hari-krishna-reddy**
`HandleGetSubOrgs` can be used to get all the Child Organizations of the current organization

### CODE REVIEW

File: shared.go [hunk 1]
```diff
@@ -784,6 +784,89 @@ func HandleGetOrg(resp http.ResponseWriter, request *http.Request) {
...
+	user, err := HandleApiAuthentication(resp, request)
+	
+	if err != nil {
+		log.Printf("[WARNING] Api authentication failed in get org: %s", err)
+		resp.WriteHeader(401)
+		resp.Write([]byte(`{"success": false}`))
+		return
+	}
+
+    orgFound:=false
```
Comment (frikky) [inline shared.go]: Indentation please :)

File: shared.go [hunk 2]
```diff
@@ -784,6 +784,89 @@ func HandleGetOrg(resp http.ResponseWriter, request *http.Request) {
...
+		  }
+		}
+	if !orgFound {
+		  log.Printf("[ERROR] User '%s' (%s) isn't a part of org %s (get)", user.Username, user.Id, orgId)
+		  resp.WriteHeader(401)
+	      resp.Write([]byte(`{"success": false, "reason": "User doesn't have access to org"}`))
+		  return
+	}
+
+	subOrgs , err = GetChildOrgs(ctx , orgId)
```
Comment (frikky) [inline shared.go]: comma usage is not SPACE COMMA SPACE

text , text2 = bad
text, text2 = good

Space after comma :))

**Comment by frikky** [comment]
Couple of general things:
- Do indentation right. Get a Golang linter.
- Be consistent.

As for the function itself, please answer:
- If we merge this, who can get access to the suborg info?
- Who SHOULD have access to suborg info? Why?
- If inside a suborg (not parent workflow), what should we show?

**Comment by satti-hari-krishna-reddy** [comment]
Access to the suborg info is determined by the authentication process. In the provided code, it checks if the authenticated user is part of the sub organization. If the user is found , they are granted access. Therefore, anyone with valid authentication credentials and is a user of the sub organization can get access to the suborg info.

File: shared.go (hunk 2 above)
Comment (satti-hari-krishna-reddy) [reply shared.go]: i think you are looking at the previous commit not the latest one : )

**Comment by satti-hari-krishna-reddy** [comment]
when there are NO sub orgs found then the code is returning an empty list for now , this is same if we are inside a suborg

**Comment by frikky** [comment]
> Access to the suborg info is determined by the authentication process. In the provided code, it checks if the authenticated user is part of the sub organization. If the user is found , they are granted access. Therefore, anyone with valid authentication credentials and is a user of the sub organization can get access to the suborg info.

To add to this, do:
- Users that are admin in the parent org can see all suborgs of the parent
- Support users can see all suborgs of the parent
- If in a suborg, make it clear what is the parent org. Show parent org only if user is part of parent org or support user

**Pull request merged**